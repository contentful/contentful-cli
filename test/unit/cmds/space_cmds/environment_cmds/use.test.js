const {
  environmentUse
} = require('../../../../../lib/cmds/space_cmds/environment_cmds/use')

const {
  createManagementClient
} = require('../../../../../lib/utils/contentful-clients')
const { setContext, getContext } = require('../../../../../lib/context')

jest.mock('../../../../../lib/utils/contentful-clients')
jest.mock('../../../../../lib/context')

const getEnvironment = jest.fn().mockResolvedValue({
  sys: {
    id: 'test'
  },
  name: 'test'
})

createManagementClient.mockReturnValue({
  getSpace: jest.fn().mockResolvedValue({
    sys: {
      id: 'mocked'
    },
    name: 'mocked',
    getEnvironment
  })
})

afterEach(() => {
  getContext.mockClear()
  setContext.mockClear()
})

test('it writes the environment id to contentfulrc.json', async () => {
  getContext.mockResolvedValue({
    managementToken: 'managementToken',
    activeSpaceId: 'spaceId'
  })
  const stubArgv = {
    context: {
      managementToken: 'managementToken',
      activeSpaceId: 'spaceId'
    },
    environmentId: 'test'
  }
  await environmentUse(stubArgv)
  expect(setContext.mock.calls[0][0]).toEqual({ activeEnvironmentId: 'test' })
})
