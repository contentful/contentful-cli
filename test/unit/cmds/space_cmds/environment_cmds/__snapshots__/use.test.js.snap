// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`active space is required 1`] = `"You need to provide a space id. You can pass it via the --space-id parameter or by running contentful space use"`;

exports[`login is required 1`] = `
"You have to be logged in to do this.
You can log in via contentful login
Or provide a managementToken via --management-Token argument"
`;
